### PR TITLE
Offering positions as an XML feed for LinkedIn's Limited Listings

### PR DESCRIPTION
**Pull request by whitbur**
Right now this is a first draft. I'd love some feedback especially in the following areas:

1. **Url structure.** Right now this code is surfaced at `/api/limited_listings`. Is there a more preferred url?
2. **Field mapping.** I did my best guessing which field mapped where, and I only filled out the required fields. Someone more familiar with the data model could do a better job there.
3. **Style and convention.** I'm not a django expert, so if I've put stuff in the wrong place please let me know. Similarly if the codebase has conventions I've missed, happy to align with those.

Here's an example of the xml output for the test fixtures data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<source>
    <lastBuildDate>Fri, 31 Jul 2020 16:55:29 UTC</lastBuildDate> 
    <publisherUrl>https://www.democracylab.org</publisherUrl>
    <publisher>DemocracyLab</publisher>
    <job>
        <company><![CDATA[DemocracyLab Test]]></company>
        <title><![CDATA[Operations]]></title>
        <description><![CDATA[HACKATHON EVENT MANAGER: DemocrayLab's hackathons are an important driver of participation in our platform. We need help to keep them running smoothly.]]></description>
        <location><![CDATA[Seattle, WA]]></location>
        <applyUrl><![CDATA[https://docs.google.com/document/d/1Skfutwl9tI3xDcAGESmJlJGIAHAKu4Czpo-CjwSxqc8/edit?usp=sharing]]></applyUrl>
        <partnerJobId><![CDATA[1]]></partnerJobId> 
    </job>
    <job>
        <company><![CDATA[Orcasound - Listen for Whales]]></company>
        <title><![CDATA[Back-End Developer]]></title>
        <description><![CDATA[Ideally familiar with Elixir/Phoenix, or excited to learn about these new tools.]]></description>
        <location><![CDATA[Redmond, WA]]></location>
        <applyUrl><![CDATA[]]></applyUrl>
        <partnerJobId><![CDATA[2]]></partnerJobId> 
    </job>
    <job>
        <company><![CDATA[Orcasound - Listen for Whales]]></company>
        <title><![CDATA[Front-End Developer]]></title>
        <description><![CDATA[Work in Elixir/Phoenix framework and UX/UI/design folks to add features via the Github Orcasite repository.]]></description>
        <location><![CDATA[Redmond, WA]]></location>
        <applyUrl><![CDATA[https://github.com/orcasound/orcasite]]></applyUrl>
        <partnerJobId><![CDATA[3]]></partnerJobId> 
    </job>
    <job>
        <company><![CDATA[Orcasound - Listen for Whales]]></company>
        <title><![CDATA[Project Manager]]></title>
        <description><![CDATA[Each Orcasound repository could use a project manager as the open-source project and its volunteer community grow.]]></description>
        <location><![CDATA[Redmond, WA]]></location>
        <applyUrl><![CDATA[http://github.com/orcasound]]></applyUrl>
        <partnerJobId><![CDATA[4]]></partnerJobId> 
    </job>
</source>
```

### CODE REVIEW

File: civictechprojects/views.py [hunk 1]
```diff
@@ -581,6 +581,40 @@ def recent_projects_list(request):
...
+        project = position.position_project
+        roleTag = Tag.get_by_name(position.position_role.first().slug)
+
+        return f"""
+        <job>
+            <company>{cdata(project.project_name)}</company>
+            <title>{cdata(roleTag.display_name)}</title>
+            <description>{cdata(position.position_description)}</description>
+            <location>{cdata(project.project_location)}</location>
+            <applyUrl>{cdata(position.description_url)}</applyUrl>
```
Comment (marlonkeating) [inline civictechprojects/views.py:602]: This position description url is sometimes blank, so we might fall back to the project profile url in that case.

File: civictechprojects/views.py [hunk 2]
```diff
@@ -581,6 +581,40 @@ def recent_projects_list(request):
...
+    def position_to_job(position):
+        project = position.position_project
+        roleTag = Tag.get_by_name(position.position_role.first().slug)
+
+        return f"""
+        <job>
+            <company>{cdata(project.project_name)}</company>
+            <title>{cdata(roleTag.display_name)}</title>
+            <description>{cdata(position.position_description)}</description>
+            <location>{cdata(project.project_location)}</location>
```
Comment (marlonkeating) [inline civictechprojects/views.py:601]: This location field is a code specific to HERE.  We'll want to use the city, state, and country values instead.

**Comment by whitbur** [comment]
> According to the XML spec, we're missing an 'Industry Code' entry for the jobs.

Huh. It's marked as required in the list up front, then optional in the appendix. No prob, although we don't have an industry field stored... How would you feel about hard coding it to `tech/Computer Software` (the value would be `4`)? [Other options are in their list here.](https://docs.microsoft.com/en-us/linkedin/shared/references/reference-tables/industry-codes)

File: civictechprojects/views.py (hunk 2 above)
Comment (whitbur) [reply civictechprojects/views.py:601]: In the spec, it's described: `This is a description of the job's location (e.g. “San Francisco, CA”). LinkedIn will automatically map this to a country and postal code unless those fields are provided. ` Do you mean the value we have stored is different from the values in `testdata.json`?

If we want to populate city/state/country, is it safe to parse the `project.project_location` field and assume it's in the US? Or did you have another source for those fields in mind?

File: civictechprojects/views.py (hunk 1 above)
Comment (whitbur) [reply civictechprojects/views.py:602]: Sure, done.

File: civictechprojects/views.py (hunk 2 above)
Comment (marlonkeating) [reply civictechprojects/views.py:601]: The testdata.json info currently in the master branch is out of date when it comes to location (see https://github.com/DemocracyLab/CivicTechExchange/pull/407 for updated test data).  To populate this now, you can use project_country, project_state, project_city.

**Comment by marlonkeating** [comment]
> How would you feel about hard coding it to `tech/Computer Software` (the value would be `4`)? [Other options are in their list here.](https://docs.microsoft.com/en-us/linkedin/shared/references/reference-tables/industry-codes)

That seems like the most appropriate option for us.

File: civictechprojects/views.py (hunk 2 above)
Comment (whitbur) [reply civictechprojects/views.py:601]: Ahh, I see. I've merged from master, makes a lot more sense now.

**Comment by whitbur** [comment]
Continued in #411, closing this one.

**Pull request closed without merging**